src/peft/tuners/lora/layer.py

```diff
@@ -917,18 +925,26 @@ def update_layer(self, adapter_name, r, lora_alpha, lora_dropout, init_lora_weig
 
         self.set_adapter(self.active_adapters)
 
+    def _get_dora_factor_view(self):
+        return (-1,) + (1,) * (self._convd + 1)
+
     def dora_init(self, adapter_name: str) -> None:
         if self.lora_magnitude_vector is None:
             # first dora layer being added, add lora_magnitude_vector to the list of learnable parameters
             self.adapter_layer_names = self.adapter_layer_names[:] + ("lora_magnitude_vector",)
 
-        dora_layer = DoraConv2dLayer(fan_in_fan_out=False)
+        dora_layer_class = self._get_dora_layer_class()
+        dora_layer = dora_layer_class(fan_in_fan_out=False)
         lora_A = self.lora_A[adapter_name].weight
         lora_B = self.lora_B[adapter_name].weight
         scaling = self.scaling[adapter_name]
         dora_layer.update_layer(base_layer=self.get_base_layer(), lora_A=lora_A, lora_B=lora_B, scaling=scaling)
         self.lora_magnitude_vector[adapter_name] = dora_layer
 
+    def _get_dora_layer_class(self) -> Type[_DoraConvNdLayer]:
```

Comment:
```suggestion
    def _get_dora_layer_class(self) -> type[_DoraConvNdLayer]:
```

This should also work.